subsys/net/ip/tcp2.c

```diff
@@ -1359,7 +1356,7 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 			}
 
 			if (len) {
```

Comment:
maybe all these if (len)  could be removed. Test the length in tcp_data_get()  
It will reduce a bit the amount of indented code blocks in these cases.

Reply:
Not really practical to do that. We would anyway need to check the length in tcp_in() in couple of places.